server/client.go

```diff
@@ -1344,7 +1344,7 @@ func (c *client) closeConnection() {
 			// Forward on unsubscribes if we are not
 			// a router ourselves.
 			if c.typ != ROUTER {
-				srv.broadcastUnSubscribe(sub)
+				srv.broadcastForceUnSubscribe(sub)
```

Comment:
I agree that https://github.com/nats-io/gnatsd/pull/551/files#diff-853eb184ac73cf9597d7833f6b89e9c9R1320 is the right place for zeroing `sub.max`, and no need new function)